prow/pod-utils/gcs/upload.go

```diff
@@ -0,0 +1,87 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package gcs
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"os"
+	"sync"
+
+	"cloud.google.com/go/storage"
+	"github.com/sirupsen/logrus"
+)
+
+// UploadFunc knows how to upload into an object
+type UploadFunc func(obj *storage.ObjectHandle) error
+
+// UploadToGcs uploads all of the data in the
+// uploadTargets map to GCS in parallel. The map is
+// keyed on GCS path under the bucket
+func UploadToGcs(bucket *storage.BucketHandle, uploadTargets map[string]UploadFunc) error {
```

Comment:
Is this logic complicated enough to justify a unit test? For example ensure that every `UploadFunc()` is called and that an error in one of them results in this returning an error